ndcube/asdf/converters/ndcube_converter.py

```diff
@@ -0,0 +1,30 @@
+from asdf.extension import Converter
+import numpy as np
+
+
+class NDCubeConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/ndcube/NDCube-*"]
+
+    @property
+    def types(self):
+        from ndcube.ndcube import NDCube
+        return [NDCube]
+
+    def select_tag(self, obj, tags, ctx):
+        # Sort the tags in reverse alphabetical order and pick the first (i.e.
+        # the one with the highest version). This assumes that all the tags for
+        # this converter are named the same other than the version number.
+        tags = list(sorted(tags, reverse=True))
+        return tags[0]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube import ndcube
+        data = np.asanyarray(node["data"])
+        wcs = node["wcs"]
+        return(ndcube.NDCube(data, wcs))
+
+    def to_yaml_tree(self, ndcube, tag, ctx):
+        node = {}
+        node["data"] = np.asarray(ndcube.data)
```

Comment:
What's the need for `asarray` here? Are there circumstances where this could lead to a data copy?

Reply:
I used `np.asarray` here also to ensure that the data attribute is consistently a `numpy.ndarray`. I used `np.asarray as a precautionary measure. We could also use `np.array`.

Reply:
To clarify my :+1:

[np.array](https://numpy.org/doc/stable/reference/generated/numpy.array.html#numpy-array) by default will copy the data. Given the call arguments here I don't expect [np.asarray](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html#numpy.asarray) will copy the data. asdf doesn't need a copy (it won't modify the data during conversion) so unless this is required by `NDCube` avoiding the copy would be more efficient.

Reply:
Yeah sure, then I suppose `np.asarray` would be a better choice.